### PR TITLE
Add partial withdrawal mandates

### PR DESCRIPTION
**Pull request by snemvalts**
* Adds partial withdrawal mandates along with mandate files
* Removes mime type from `MandateContentFile` – `text/html` is default
* Removes redundant details setting in `MandateFactories`

### CODE REVIEW

File: src/main/java/ee/tuleva/onboarding/epis/mandate/details/PartialWithdrawalMandateDetails.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  @NotNull private final List<FundWithdrawalAmount> fundWithdrawalAmounts;
+
+  @Valid3LetterCountryCode private final String taxResidency;
+
+  @JsonCreator
+  public PartialWithdrawalMandateDetails(
+      @JsonProperty("pillar") Pillar pillar,
+      @JsonProperty("bankAccountDetails") BankAccountDetails bankAccountDetails,
+      @JsonProperty("fundWithdrawalAmounts") List<FundWithdrawalAmount> fundWithdrawalAmounts,
+      @JsonProperty("taxResidency") String taxResidency) {
```
Comment (ErkoRisthein) [inline src/main/java/ee/tuleva/onboarding/epis/mandate/details/PartialWithdrawalMandateDetails.java:26]: I think it's just "residency"

File: src/test/groovy/ee/tuleva/onboarding/mandate/GenericMandateIntegrationTest.java [hunk 2]
```diff
@@ -65,7 +65,8 @@ static Stream<Arguments> testMandateDetails() {
                 SECOND,
                 FundPensionOpeningMandateDetails.FundPensionFrequency.MONTHLY,
                 new FundPensionOpeningMandateDetails.FundPensionDuration(20, false),
-                new BankAccountDetails(ESTONIAN, BankAccountDetails.Bank.LHV, "EE_TEST_IBAN"))));
+                new BankAccountDetails(ESTONIAN, BankAccountDetails.Bank.LHV, "EE_TEST_IBAN"))),
```
Comment (ErkoRisthein) [inline src/test/groovy/ee/tuleva/onboarding/mandate/GenericMandateIntegrationTest.java:68]: You can use a static import for the bank

File: src/test/groovy/ee/tuleva/onboarding/mandate/MandateFixture.groovy [hunk 3]
```diff
@@ -31,6 +32,20 @@ import static ee.tuleva.onboarding.user.address.AddressFixture.addressFixture
...
 
+  public static PartialWithdrawalMandateDetails aPartialWithdrawalMandateDetails = new PartialWithdrawalMandateDetails(SECOND,
+      new BankAccountDetails(ESTONIAN, BankAccountDetails.Bank.fromIban("EE3477123123123"), "EE_TEST_IBAN"),
+      List.of(new PartialWithdrawalMandateDetails.FundWithdrawalAmount("EE3600109435", 10, BigDecimal.valueOf(20)),
+          new PartialWithdrawalMandateDetails.FundWithdrawalAmount("EE3600019766", 5, BigDecimal.valueOf(30))),
+      "EST"
+  )
+
+  public static PartialWithdrawalMandateDetails aThirdPillarPartialWithdrawalMandateDetails = new PartialWithdrawalMandateDetails(THIRD,
+      new BankAccountDetails(ESTONIAN, BankAccountDetails.Bank.fromIban("EE3477123123123"), "EE_TEST_IBAN"),
```
Comment (ErkoRisthein) [inline src/test/groovy/ee/tuleva/onboarding/mandate/MandateFixture.groovy:43]: Static import `Bank`

File: src/test/groovy/ee/tuleva/onboarding/mandate/GenericMandateIntegrationTest.java (hunk 2 above)
Comment (snemvalts) [reply src/test/groovy/ee/tuleva/onboarding/mandate/GenericMandateIntegrationTest.java:68]: Should see if there is a spotless setting for it

**Pull request merged**